en/test_and_ci/unit_tests.md

```diff
@@ -1,11 +1,34 @@
 # Unit Tests
 
-PX4 provides a simple base [Unittest-class](https://github.com/PX4/Firmware/blob/master/src/include/unit_test.h). Each developer is encouraged to write unit tests in the process of adding a new feature to the PX4 framework.
-
-## Writing a Test
-
-1. Create a new .cpp file within [tests](https://github.com/PX4/Firmware/tree/master/src/systemcmds/tests) with name **test_[description].cpp**. 
-1. Within **test_[description].cpp** include the base unittest-class `<unit_test.h>` and all files required to write a test for the new feature. 
+PX4 provides several methods for writing unit tests. Each developer is encouraged to write unit tests during all parts of development, including adding new features, fixing bugs, and refactoring.
+
+1. Unit tests with GTest - tests that have minimal, internal-only dependencies
+1. Functional tests with GTest - tests that depend on parameters and uORB messages
+1. SITL unit tests. This is for tests that need to run in full SITL. These tests are much slower to run and harder to debug, so it is recommended to use GTest instead when possible.
+
+## Writing a GTest Unit Test
+
+1. Unit tests should be arranged in three sections: setup, run, check results. Each test should test one very specific behavior or setup case, so if a test fails it is obvious what is wrong. Please try to follow these standards when possible.
+1. Copy and rename the [example unit test](https://github.com/PX4/Firmware/blob/master/src/modules/mc_att_control/AttitudeControl/AttitudeControlTest.cpp) to the directory the code to be tested is in.
+1. Add the new file to the directory's `CMakeLists.txt`. It should look something like `px4_add_unit_gtest(SRC MyNewUnitTest.cpp LINKLIBS <library_to_be_tested>)`
+1. Add the desired test functionality. This will mean including the header files required for your specific tests, adding new tests (each with an individual name) and putting the logic for the setup, running the code to be tested and verifying that it behaves as expected.
+1. In general, if you need access to advanced GTest utilities, data structures from the STL or need to link to `parameters` or `uorb` libraries you should use the functional tests instead.
+1. Tests can be run via `make tests`, after which you will find the binary in `build/px4_sitl_test/unit-MyNewUnit`. It can be run directly in a debugger.
+
+## Writing a GTest Functional Test
+
+1. In general (and similar to unit tests), functional tests should be arranged in three sections: setup, run, check results. Each test should test one very specific behavior or setup case, so if a test fails it is obvious what is wrong. Please try to follow these standards when possible.
+1. Copy and rename the [example functional test](https://github.com/PX4/Firmware/blob/master/src/lib/parameters/ParameterTest.cpp) to the directory the code to be tested is in.
```

Comment:
Again, it is great to use a real test as an example, but best to refer to it so that is obvious - otherwise the general assumption might be that you are pointing to an example template suitable for generic copy/paste.

Reply:
To some degree this was written as an example, since it is assumed that most functional tests will need to set/get parameters and sub/pub uORB messages. But I'll make it more clear.